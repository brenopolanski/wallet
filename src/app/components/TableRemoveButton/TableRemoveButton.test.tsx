import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";

import { TableRemoveButton } from "./TableRemoveButton";

describe("TableRemoveButton", () => {
	it("should render", () => {
		const { container } = render(<TableRemoveButton onClick={jest.fn()} />);

		expect(screen.getByTestId("TableRemoveButton")).toHaveTextContent("trash.svg");

		expect(container).toMatchSnapshot();
	});

	it("should render compact", () => {
		const { container } = render(<TableRemoveButton onClick={jest.fn()} isCompact />);

		expect(screen.getByTestId("TableRemoveButton")).toHaveTextContent("trash.svg");

		expect(container).toMatchSnapshot();
	});

	it("should execute onClick callback", () => {
		const onClick = jest.fn();

		render(<TableRemoveButton onClick={onClick} />);

		fireEvent.click(screen.getByTestId("TableRemoveButton"));

		expect(onClick).toHaveBeenCalledWith(expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }));
	});
});
