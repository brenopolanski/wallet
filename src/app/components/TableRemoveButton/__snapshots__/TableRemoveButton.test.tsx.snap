// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRemoveButton should render 1`] = `
<div>
  <button
    class="OriginalButton__StyledButton-sc-o0fs2j-0 fxPCrX"
    data-testid="TableRemoveButton"
    type="button"
  >
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          trash.svg
        </svg>
      </div>
    </div>
  </button>
</div>
`;

exports[`TableRemoveButton should render compact 1`] = `
<div>
  <button
    class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF text-theme-danger-400 hover:text-theme-danger-500"
    data-testid="TableRemoveButton"
    type="button"
  >
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="sc-bdvvtL fCrhkg"
        height="16"
        width="16"
      >
        <svg>
          trash.svg
        </svg>
      </div>
    </div>
  </button>
</div>
`;
